airbyte-integrations/connectors/destination-bigquery/src/main/java/io/airbyte/integrations/destination/bigquery/formatter/BigQueryRecordFormatter.java

```diff
@@ -4,82 +4,59 @@
-import io.airbyte.protocol.models.v0.AirbyteRecordMessage;
+import io.airbyte.cdk.integrations.destination.async.model.PartialAirbyteRecordMessage;
+import io.airbyte.commons.json.Jsons;
 import java.util.HashMap;
-import java.util.HashSet;
-import java.util.Map;
-import java.util.Set;
+import java.util.UUID;
+import java.util.concurrent.TimeUnit;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 /**
  * The class formats incoming JsonSchema and AirbyteRecord in order to be inline with a
  * corresponding uploader.
  */
-public abstract class BigQueryRecordFormatter {
+public class BigQueryRecordFormatter {
 
+  public static final Schema SCHEMA_V2 = Schema.of(
+      Field.of(JavaBaseConstants.COLUMN_NAME_AB_RAW_ID, StandardSQLTypeName.STRING),
+      Field.of(JavaBaseConstants.COLUMN_NAME_AB_EXTRACTED_AT, StandardSQLTypeName.TIMESTAMP),
+      Field.of(JavaBaseConstants.COLUMN_NAME_AB_LOADED_AT, StandardSQLTypeName.TIMESTAMP),
+      Field.of(JavaBaseConstants.COLUMN_NAME_DATA, StandardSQLTypeName.STRING));
   private static final Logger LOGGER = LoggerFactory.getLogger(BigQueryRecordFormatter.class);
 
-  protected Schema bigQuerySchema;
-  protected final Map<String, Set<String>> mapOfFailedFields = new HashMap<>();
   protected final StandardNameTransformer namingResolver;
-  protected final JsonNode originalJsonSchema;
-  protected JsonNode jsonSchema;
 
-  /**
-   * These parameters are required for the correct operation of denormalize version of the connector.
-   */
-  protected final Set<String> invalidKeys = new HashSet<>();
-  protected final Set<String> fieldsContainRefDefinitionValue = new HashSet<>();
-
-  public BigQueryRecordFormatter(final JsonNode jsonSchema, final StandardNameTransformer namingResolver) {
+  public BigQueryRecordFormatter(final StandardNameTransformer namingResolver) {
     this.namingResolver = namingResolver;
-    this.originalJsonSchema = jsonSchema.deepCopy();
-    this.jsonSchema = formatJsonSchema(jsonSchema.deepCopy());
   }
 
-  protected JsonNode formatJsonSchema(final JsonNode jsonSchema) {
-    // Do nothing by default
-    return jsonSchema;
-  };
-
-  public abstract JsonNode formatRecord(AirbyteRecordMessage recordMessage);
-
   public String formatRecord(PartialAirbyteMessage recordMessage) {
-    return "";
-  }
-
-  public Schema getBigQuerySchema() {
-    if (bigQuerySchema == null) {
-      bigQuerySchema = getBigQuerySchema(jsonSchema);
-    }
-    return bigQuerySchema;
+    // Map.of has a @NonNull requirement, so creating a new Hash map
+    final HashMap<String, Object> destinationV2record = new HashMap<>();
+    destinationV2record.put(JavaBaseConstants.COLUMN_NAME_AB_RAW_ID, UUID.randomUUID().toString());
+    destinationV2record.put(JavaBaseConstants.COLUMN_NAME_AB_EXTRACTED_AT, getEmittedAtField(recordMessage.getRecord()));
+    destinationV2record.put(JavaBaseConstants.COLUMN_NAME_AB_LOADED_AT, null);
```

Comment:
is there even a point to sending an explicit null for loaded_at? (I'm guessing this is copied from existing code, just seems weird)

... or, if we're immediately serializing it, why not just build a `JsonNode` directly?